modules/odata/odata-core/src/main/java/org/eclipse/dirigible/engine/odata2/sql/processor/AbstractSQLProcessor.java

```diff
@@ -330,11 +331,17 @@ public ODataResponse createEntity(final PostUriInfo uriInfo, final InputStream c
             updateCreateEventHandlerContext(handlerContext, connection, uriInfo, entry);
             this.odata2EventHandler.beforeCreateEntity(uriInfo,
                     requestContentType, contentType, entry, handlerContext);
+            if (isErrorDocumentPresent(handlerContext)) {
+                return (ODataResponse) handlerContext.get(ERROR_DOCUMENT);
+            }

```

Comment:
getting the result of executing method from input param of the method is bad. Why not just to return ODataResponse from the handler method?